### ISSUE TITLE
Autoscaler is not using internal cluster address

### ISSUE THREAD

**Issue by prabalsharma** [cloudfoundry-incubator/kubecf#870]
**Describe the bug**
Autoscaler is not using internal cluster address. Instead it expects an external dns entry https://github.com/cloudfoundry-incubator/kubecf/blob/master/deploy/helm/kubecf/assets/operations/instance_groups/app-autoscaler.yaml#L167

**To Reproduce**
enable autoscaler on cluster with lb=true

**Expected behavior**

```
logs 2020/05/13 17:49:27, /var/vcap/sys/log/operator/operator.stdout.log, {"data":{"API":"https://api.ci-gke-643ccc42910026da.ci.kubecf.charmedquarks.me","error":"Get https://api.ci-gke-643ccc42910026da.ci.kubecf.charmedquarks.me/v2/info: dial tcp: │
│  lookup api.ci-gke-643ccc42910026da.ci.kubecf.charmedquarks.me on 10.19.249.66:53: no such host"},"log_level":2,"log_time":"2020-05-13T17:49:27Z","message":"operator.failed to login cloud foundry","source":"operator","timestamp":"1589392167.5512430 │
│ 67"}
```

**Environment**

```
kubecf-2.0-RC1                  2.0-RC1 
cf-operator-4.5.6+0.gffc6f942   4.5.6+0.gffc6f942
```

**Additional context**

**Comment by satadruroy** [comment]
We were able to get the autoscaler pods to come up by hacking https://github.com/cloudfoundry-incubator/kubecf/blob/master/deploy/helm/kubecf/assets/operations/instance_groups/app-autoscaler.yaml#L167 to point to `https://cloud-controller-ng.service.cf.internal:9024` but it seems CC was still trying to connect to UAA at the public endpoint and was not able to do so until the DNS propagation was complete:

```{"data":{"API":"https://cloud-controller-ng.service.cf.internal:9024","error":"Post https://uaa.sr.cap2.susecap.biz/oauth/token: dial tcp: lookup uaa.sr.cap2.susecap.biz on 10.0.122.26:53: no such host"},```

**Comment by jandubois** [comment]
I've looked at your cluster the next day, and I could not find the error in any of the logs (I pulled logs from *all* containers using `klog`).

```
$ ag uaa.sr.cap2.susecap.biz/oauth/token
routing-api-1/job/routing-api-routing-api/kube.log
5:{"timestamp":"2020-05-14T23:51:58.952907026Z","level":"info","source":"routing-api","message":"routing-api.received-issuer","data":{"issuer":"https://uaa.sr.cap2.susecap.biz/oauth/token"}}

routing-api-0/job/routing-api-routing-api/kube.log
5:{"timestamp":"2020-05-14T23:51:41.496995111Z","level":"info","source":"routing-api","message":"routing-api.received-issuer","data":{"issuer":"https://uaa.sr.cap2.susecap.biz/oauth/token"}}
```

I see that both `api-0` and `api-1` have been restarted once, so I assume the logs are lost now. Please capture logs during the setup before configuring DNS to see if that sheds more light on it.

**Comment by satadruroy** [comment]
This can be closed with #891 above. Opened a separate issue for UAA external DNS resolution failure #897. For now, we should document operators should setup DNS as soon as the LB service(s) have external IPs plumbed on them (which happens fairly quickly right after the deployment starts on hyperscaler platforms). It is expected that by the time the autoscaler pods come up DNS propagation would be complete and #897 can be avoided.
